cms/djangoapps/contentstore/views/import_export.py

```diff
@@ -349,75 +469,50 @@ def export_handler(request, course_key_string):
-            unit = None
-            failed_item = None
-            parent = None
-            try:
-                failed_item = modulestore().get_item(exc.location)
-                parent_loc = modulestore().get_parent_location(failed_item.location)
-
-                if parent_loc is not None:
-                    parent = modulestore().get_item(parent_loc)
-                    if parent.location.category == 'vertical':
-                        unit = parent
-            except:  # pylint: disable=bare-except
-                # if we have a nested exception, then we'll show the more generic error message
-                pass
-
-            return render_to_response('export.html', {
-                'context_course': course_module,
-                'in_err': True,
-                'raw_err_msg': str(exc),
-                'failed_module': failed_item,
-                'unit': unit,
-                'edit_unit_url': reverse_usage_url("container_handler", parent.location) if parent else "",
-                'course_home_url': reverse_course_url("course_handler", course_key),
-                'export_url': export_url
-            })
-        except Exception as exc:
-            log.exception('There was an error exporting course %s', course_module.id)
-            return render_to_response('export.html', {
-                'context_course': course_module,
-                'in_err': True,
-                'unit': None,
-                'raw_err_msg': str(exc),
-                'course_home_url': reverse_course_url("course_handler", course_key),
-                'export_url': export_url
-            })
-        finally:
-            shutil.rmtree(root_dir / name)
-
-        wrapper = FileWrapper(export_file)
-        response = HttpResponse(wrapper, content_type='application/x-tgz')
-        response['Content-Disposition'] = 'attachment; filename=%s' % os.path.basename(export_file.name.encode('utf-8'))
-        response['Content-Length'] = os.path.getsize(export_file.name)
-        return response
+            tarball = create_export_tarball(course_module, course_key, context, library=library)
+        except SerializationError:
+            return render_to_response(template, context)
+        return send_tarball(tarball)
 
     elif 'text/html' in requested_format:
-        return render_to_response('export.html', {
-            'context_course': course_module,
-            'export_url': export_url
-        })
+        return render_to_response(template, context)
 
     else:
         # Only HTML or x-tgz request formats are supported (no JSON).
         return HttpResponse(status=406)
+
+
+# pylint: disable=unused-argument
+@ensure_csrf_cookie
+@login_required
+@require_http_methods(("GET",))
+def library_export_handler(request, library_key_string):
+    """
+    Library export has enough separate concerns that its functionality is not
+    easily piggy-backed onto the course export function.
```

Comment:
Is this comment out of date? Doesn't seem to jive with the fact that this method just calls the course method.